src/openms/source/CHEMISTRY/AASequence.cpp

```diff
@@ -176,11 +178,16 @@ namespace OpenMS
 
   EmpiricalFormula AASequence::getFormula(Residue::ResidueType type, Int charge) const
   {
-    EmpiricalFormula ef;
+
+    // x/c-ion for a single residue is not defined
+    if (peptide_.size() == 1 && (type == Residue::XIon || type == Residue::CIon))
```

Comment:
So,
size==1 throws an Exception, whereas
size==0 returns an empty EF.
Would it make sense to check for >=1 ?